### DISCUSSION TITLE
picker 的 columns 要怎么定义类型才行

### DISCUSSION THREAD

**Discussion by tlerbao** [youzan/vant#11175] [Q&A]
<img width="1193" alt="image" src="https://user-images.githubusercontent.com/20442232/197924136-aca46443-d4a0-48c3-b1eb-658f76306625.png">

要怎么写才不会报错呢？

**Comment by wjw-gavin** [comment]
![image](https://user-images.githubusercontent.com/19986739/198173956-93d0dbc9-f7d9-4342-bf38-466f1c44e142.png)
![image](https://user-images.githubusercontent.com/19986739/198174279-a5b74c50-e200-4b50-abfd-82570bcfb51e.png)

可以贴一下你的数据结构看下

**Reply by tlerbao** [reply]
@wjw-gavin 就是个数组啊，我有三列数据

<img width="781" alt="image" src="https://user-images.githubusercontent.com/20442232/198248577-2cee1556-63c4-4937-a2ae-b7c2153ca573.png">

**Comment by wjw-gavin** [comment]
> <img alt="image" width="1193" src="https://user-images.githubusercontent.com/20442232/197924136-aca46443-d4a0-48c3-b1eb-658f76306625.png">
> 
> 要怎么写才不会报错呢？

。。。。
类型是数组 改成 `const pickData = ref<PickerOption[]>([])`试试，类型定义错了`<PickerOption[]>`

**Reply by wjw-gavin** [reply]
类型后跟`[]`才是定义数组类型

**Reply by tlerbao** [reply]
不行：let pickerData = ref<PickerOption[] | PickerOption[][]>([]) 
不行：let pickerData = ref<PickerOption[]>([])
@wjw-gavin

**Reply by wjw-gavin** [reply]
我这样写的，我这没报错....

**Reply by tlerbao** [reply]
找到问题 unplugin-auto-import 的问题，用自动导入 的ref()就会报错，import { ref } from 'vue' 后不报错了
